src/api/app/components/notification_component.rb

```diff
@@ -29,4 +29,65 @@ def notification_icon
       tag.i(class: ['fas', NOTIFICATION_ICON[@notification.notifiable_type]], title: NOTIFICATION_TITLE[@notification.notifiable_type])
     end
   end
+
+  def description
+    case @notification.event_type
+    when 'Event::ReportForUser'
+      description_for_user_report
+    when 'Event::ReportForComment'
+      description_for_comment_report
+    else
+      @notification.description
+    end
+  end
```

Comment:
We just moved all of this into the model. Why are you moving it out of it again?

Reply:
Because this only generating view related data, links and html badges. I am not sure this new code belongs to model.